src/service/cia301/co_csdo.c

```diff
@@ -635,7 +629,7 @@ static CO_ERR COCSdoDownloadSubBlock       (CO_CSDO *csdo)
             cmd = SET_BITS( BLOCK_DOWNLOAD_CMD_CCS,                 
                             BLOCK_DOWNLOAD_CMD_SCS_CCS_BIT_OFFSET,  
                             BLOCK_DOWNLOAD_CMD_SCS_CCS_BIT_MASK);
-            cmd |= SET_BITS(BLOCK_DOWNLOAD_FRM_SUBBLOCK_REQUEST_SEGDATA_BYTE_SIZE - block->BytesInLastSeg + 1,   
+            cmd |= SET_BITS(BLOCK_DOWNLOAD_FRM_SUBBLOCK_REQUEST_SEGDATA_BYTE_SIZE - block->BytesInLastSeg,   
```

Comment:
this seems to be the main functional difference in this PR.

Reply:
Added a comment as this could be hard to spot in the future

Reply:
Thank you!